core/arch/arm/pta/crypto_aux.c

```diff
@@ -0,0 +1,75 @@
+/* SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) 2018, Linaro Limited
+ */
+#include <kernel/msg_param.h>
+#include <kernel/pseudo_ta.h>
+#include <kernel/user_ta.h>
+#include <pta_crypto_aux.h>
+#include <tee/tee_cryp_utl.h>
+
+static TEE_Result crypto_reseed(struct tee_ta_session *s __unused,
+				uint32_t param_types,
+				TEE_Param params[TEE_NUM_PARAMS])
+{
+	size_t entropy_sz;
+	uint8_t *entropy_input;
+	uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+					  TEE_PARAM_TYPE_NONE,
+					  TEE_PARAM_TYPE_NONE,
+					  TEE_PARAM_TYPE_NONE);
+
+	if (exp_pt != param_types)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	entropy_input = params[0].memref.buffer;
+	entropy_sz = params[0].memref.size;
+	if (!entropy_sz)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	return tee_prng_add_entropy(entropy_input, entropy_sz);
+}
+
+static TEE_Result open_session(uint32_t param_types __unused,
+			       TEE_Param params[TEE_NUM_PARAMS] __unused,
+			       void **sess_ctx __unused)
+{
+	struct tee_ta_session *s;
+
+	/* Check that we're called from a user TA */
+	s = tee_ta_get_calling_session();
```

Comment:
why forbidding non-secure to feed entropy?

Reply:
The purpose of this pTA is to extend GP Core API, which doesn't define any call for re-seeding entropy pool.

@jenswi-linaro could you please share your thoughts regarding non-secure calls to this pTA?

Reply:
This PTA is only supposed to be called from a secure TA. Actually I wouldn't mind if there was an extra parameter to the call indicating if the origin of the entropy is secure or non-secure even if we will ignore that for now.